pkg/driver/node.go

```diff
@@ -359,3 +379,56 @@ func (d *Driver) nodePublishVolumeForFileSystem(req *csi.NodePublishVolumeReques
 
 	return nil
 }
+
+// findDevicePath finds and verifies the path of device
+// if the device is not nvme device, return the path
+// if the device is nvme device, returns the actual device path
+func (d *Driver) findDevicePath(devicePath, volumeId string) (string, error) {
+	exists, err := d.mounter.ExistsPath(devicePath)
+	if err != nil {
+		return "", err
+	}
+
+	// If the path exists, it is normal device
+	if exists {
+		return devicePath, nil
+	}
+
+	// Else find the nvme device path using volume ID
+	// This is the magic name on which AWS presents NVME devices under /dev/disk/by-id/
+	// For example, vol-0fab1d5e3f72a5e23 creates a symlink at /dev/disk/by-id/nvme-Amazon_Elastic_Block_Store_vol0fab1d5e3f72a5e23
+	nvmeName := "nvme-Amazon_Elastic_Block_Store_" + strings.Replace(volumeId, "-", "", -1)
+
+	return findNvmeVolume(nvmeName)
+}
+
+// findNvmeVolume looks for the nvme volume with the specified name
+// It follows the symlink (if it exists) and returns the absolute path to the device
+func findNvmeVolume(findName string) (device string, err error) {
+	p := filepath.Join("/dev/disk/by-id/", findName)
+	stat, err := os.Lstat(p)
+	if err != nil {
+		if os.IsNotExist(err) {
+			klog.V(5).Infof("nvme path %q not found", p)
+			return "", nil
```

Comment:
Should this return an error? The condition of `""` isn't checked in the previous functions.